src/dev/create-extension.ts

```diff
@@ -37,13 +37,39 @@ const getProjectPath = (rawArgs: string[]) => {
   return { projectPath };
 };
 
-const getChangedFilesFromFirstCommit = async (projectPath: string): Promise<string[]> => {
+const getDiffFilesFromFirstCommit = async ({
+  projectPath,
+  isDeleted = false,
+}: {
+  projectPath: string;
+  isDeleted?: boolean;
+}): Promise<string[]> => {
+  if (isDeleted) {
+    // all files that have ever existed in the repo's history
+    const { stdout: allFiles } = await execa(
+      "git",
+      ["log", "--all", "--diff-filter=ACDMRT", "--name-only", "--format="],
+      { cwd: projectPath },
+    );
+
+    const { stdout: currentFiles } = await execa("git", ["ls-files"], {
+      cwd: projectPath,
+    });
+
+    const allFilesSet = new Set(allFiles.split("\n").filter(Boolean));
```

Comment:
nit: it's better to use just array here since you're using `Array.from(allFilesSet)` anyway

Reply:
```ts
  const { stdout: allFiles } = await execa(
    "git",
    ["log", "--all", "--diff-filter=ACDMRT", "--name-only", "--format="],
    { cwd: projectPath },
  );
```

Sometimes `allFiles` can have duplicates, by using set here we make we don't add them again and have only one path. 

Changed it for `currentFiles` and using array there 🙌

Reply:
> Sometimes allFiles can have duplicates, by using set here we make we don't add them again and have only one path.

Thank you for the clarification, didn't think about it!

> Changed it for currentFiles and using array there 🙌

For me, Set is good for `currentFiles` since `.has` is much faster than `.includes`, but we can notice it only for a large amounts of files, so I think for us there's no difference